app/lib/submission_builder/ty2022/states/authentication_header.rb

```diff
@@ -16,8 +16,40 @@ def document
                 xml.NoFinancialProduct "X"
               end
             end
-            xml.PrimDrvrLcnsOrStateIssdIdGrp do
-              xml.DoNotHaveDrvrLcnsOrStIssdId "X"
+            if @submission.data_source.state_name == "New York"
+              xml.PrimDrvrLcnsOrStateIssdIdGrp do
+                state_id = @submission.data_source.primary_state_id
+                if state_id.present? && (state_id.id_type_driver_license? || state_id.id_type_dmv_bmv?)
+                  xml.DrvrLcnsNum state_id.id_number
+                  xml.DrvrLcnsStCd state_id.state
+                  xml.DrvrLcnsExprDt do
+                    xml.ExprDt state_id.expiration_date.strftime("%Y-%m-%d")
+                  end
+                  xml.DrvrLcnsIssueDt state_id.issue_date.strftime("%Y-%m-%d")
+                else
+                  xml.DoNotHaveDrvrLcnsOrStIssdId "X"
+                end
```

Comment:
since these blocks are identical, you could probably reuse the code somehow, maybe through a helper method?

```
xml.PrimDrvrLcnsOrStateIssdIdGrp do
  state_id = @submission.data_source.primary_state_id
  fill_in_state_id(xml, state_id)
end

if @submission.data_source.filing_status_mfj?
  xml.SpsDrvrLcnsOrStateIssdIdGrp do
    state_id = @submission.data_source.spouse_state_id
    fill_in_state_id(xml, state_id)
  end
end

def fill_in_state_id(xml, state_id)
  if state_id.present? && (state_id.id_type_driver_license? || state_id.id_type_dmv_bmv?)
    xml.DrvrLcnsNum state_id.id_number
    xml.DrvrLcnsStCd state_id.state
    xml.DrvrLcnsExprDt do
      xml.ExprDt state_id.expiration_date.strftime("%Y-%m-%d")
    end
    xml.DrvrLcnsIssueDt state_id.issue_date.strftime("%Y-%m-%d")
  else
    xml.DoNotHaveDrvrLcnsOrStIssdId "X"
  end
end
```

Reply:
not necessary though, up to you!